pkg/manifests/manifests_test.go

```diff
@@ -1599,6 +1601,131 @@ alertmanagerMain:
 	}
 }
 
+func TestK8sPrometheusAdapterAuditLog(t *testing.T) {
+	// setting audit level should add the audit args
+	tt := []struct {
+		config       string
+		expectedArgs []string
+		err          error
+	}{{
+		config: ``,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/metadata-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+	}, {
+		config: `k8sPrometheusAdapter: `,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/metadata-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+	}, {
+		config: `
+k8sPrometheusAdapter:
+  audit: {} `,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/metadata-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+	}, {
+		config: `
+k8sPrometheusAdapter:
+  audit:
+    profile: Request
+`,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/request-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+		err: nil,
+	}, {
+		config: `
+k8sPrometheusAdapter:
+  audit:
+    profile: RequestResponse
+`,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/requestresponse-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+		err: nil,
+	}, {
+		config: `
+  k8sPrometheusAdapter:
+    audit:
+     profile: None
+`,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/none-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+		err: nil,
+	}, {
+		config: `
+  k8sPrometheusAdapter:
+    audit:
+      profile: Foobar  # should generate an error
+`,
+		err: ErrConfigValidation,
+	}}
+
+	for _, test := range tt {
+		c, err := NewConfigFromString(test.config)
+		if err != nil {
+			t.Logf("%s\n\n", test.config)
+			t.Fatal(err)
+		}
+		f := NewFactory("openshift-monitoring", "openshift-user-workload-monitoring", c,
+			defaultInfrastructureReader(), &fakeProxyReader{}, NewAssets(assetsPath))
+
+		d, err := f.PrometheusAdapterDeployment("foo", map[string]string{
+			"requestheader-allowed-names":        "",
+			"requestheader-extra-headers-prefix": "",
+			"requestheader-group-headers":        "",
+			"requestheader-username-headers":     "",
+		})
+
+		if test.err != nil || err != nil {
+			// fail only if the error isn't what is expected
+			if !errors.Is(err, test.err) {
+				t.Logf("%s\n\n", test.config)
+				t.Fatalf("Expected error %q but got %q", test.err, err)
+			}
+			break
+		}
+
+		adapterArgs := d.Spec.Template.Spec.Containers[0].Args
+		auditArgs := []string{}
+		for _, arg := range adapterArgs {
+			if strings.HasPrefix(arg, "--audit-") {
```

Comment:
Could we check only `--audit-policy-file` here? That would avoid checking the values of `--audit-log-path` and `--audit-log-maxsize` everytime even though they are not configurable by the users.

Reply:
we could but I will keep at least one check since something needs to validate the args are set properly